### PR TITLE
Feature/added filter by category

### PR DESCRIPTION
**Pull request by angarita-dev**
# Hi! :rocket: 

This it the pull request for the milestone #4 of the Bookstore project.

### Added features:
* `filter` reducer.
* `CHANGE_FILTER` action.
* `CategoryFilter` component.
* Filtered books using current filter category.

@codingAngarita :wave:

### CODE REVIEW

File: src/actions/reducers/book.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { CREATE_BOOK, REMOVE_BOOK } from '../index';
+
+export const generateId = () => Math.floor(Math.random() * (99999 - 10000) + 10000);
```
Comment (bellom) [inline src/actions/reducers/book.js]: **[Optional]**
Kindly be aware that you can use uuidv4 to automatically generate ids. read more 👉 https://www.npmjs.com/package/uuidv4

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Book(props) {
+  const { id } = props;
+  const { title } = props;
+  const { category } = props;
+  const { removeBook } = props;
```
Comment (bellom) [inline src/components/Book.js]: We can simply do this in one line;
`const { id, title, .... } = props`

File: src/containers/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// Components
+import CategoryFilter from '../components/CategoryFilter';
+import Book from '../components/Book';
+
+const BooksList = props => {
+  const { books } = props;
+  const { removeBook } = props;
+  const { changeFilter } = props;
+  const { categoryList } = props;
+  const { currentFilter } = props;
```
Comment (bellom) [inline src/containers/BooksList.js]: Let's do this in one line 👍

File: src/containers/BooksForm.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        </select>
+        <span
+          onClick={this.handleSubmit}
+          onKeyPress={this.handleSubmit}
+          role="button"
+          tabIndex="0"
+        >
+          Save book
+        </span>
+      </div>
```
Comment (bellom) [inline src/containers/BooksForm.js:81]: Let's validate our form so that users can't create an empty book.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @codingAngarita  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by angarita-dev** [comment]
# Hi! :robot: 

### New changes:
* Added `uuid` package to manage book ids.
* Used one line prop destructure
* Added empty book title validation before saving

**Review by bellom — APPROVED** [review]
## Hi @codingAngarita  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**